src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/ReferralEventPublisher.kt

```diff
@@ -32,6 +34,7 @@ class ReferralEventPublisher(
   }
 
   fun referralConcludedEvent(referral: Referral, eventType: ReferralEventType) {
+    referral.currentAssignee ?: logger.warn("Concluding referral has no current assignment ${referral.id} for event type $eventType")
```

Comment:
This message feels like knowing how the published message is going to be used. Should this be on the listener side?

Reply:
Alternatively, this feels like a kind of "this instance of generic data has more constraints" which feel like calls for a separate event class

Reply:
Not a blocker though -- the reasons in the PR (which I read too late, sorry) are sound